traversal/procedure/VertexProcedure.java

```diff
@@ -73,12 +76,19 @@ public FunctionalIterator<VertexMap> iterator(GraphManager graphMgr, Traversal.P
         LOG.trace(params.toString());
         LOG.trace(this.toString());
         assert vertex.id().isRetrievable() && modifiers.filter().variables().contains(vertex.id().asVariable().asRetrievable());
-        FunctionalIterator<? extends Vertex<?, ?>> iterator = vertex.iterator(graphMgr, params, modifiers.sorting().order(vertex.id()));
+        Optional<Order> order = modifiers.sorting().order(vertex.id());
+        FunctionalIterator<? extends Vertex<?, ?>> iterator = vertex.iterator(graphMgr, params, order.orElse(ASC), order.isPresent());
         for (ProcedureEdge<?, ?> e : vertex.loops()) {
             iterator = iterator.filter(v -> e.isClosure(graphMgr, v, v, params));
         }
 
-        return iterator.map(v -> VertexMap.of(map(pair(vertex.id().asVariable().asRetrievable(), v))));
+        return iterator.map(v -> {
+            if (v.isThing() && v.asThing().isAttribute() && v.asThing().asAttribute().isValue()) {
```

Comment:
because some of the vertices may represent `Value` vertices, we have to unwrap them and return just `Attribute` vertices as is expected by all layers above the traversal engine.

Reply:
this bit is a tad ugly, would be good to find a better place for it